cmd/agent/api/agent/agent.go

```diff
@@ -278,6 +281,40 @@ func getRuntimeConfig(w http.ResponseWriter, r *http.Request) {
 	w.Write(runtimeConfig)
 }
 
+func setRuntimeConfig(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	setting := vars["setting"]
+	log.Info("Got a request to change a setting: %s", setting)
+	r.ParseForm()
+	value := html.UnescapeString(r.Form.Get("value"))
+
+	if _, ok := config.RuntimeSettings[setting]; !ok {
+		body, _ := json.Marshal(map[string]string{"error": "unknown setting"})
+		http.Error(w, string(body), 500)
+		return
+	}
+	if err := config.RuntimeSettings[setting].Set(value); err != nil {
+		body, _ := json.Marshal(map[string]string{"error": err.Error()})
+		http.Error(w, string(body), 500)
+		return
+	}
+}
+
+func getRuntimeConfigurableSettings(w http.ResponseWriter, r *http.Request) {
+	configurableSettings := make(map[string]string)
+	for name, setting := range config.RuntimeSettings {
+		configurableSettings[name] = setting.Description()
+	}
```

Comment:
Same here, we should have a function to get those: `config.getRuntimeSettings() -> map[string]string` ?

Reply:
Will do, cf. above, it will better isolate the runtime settings stuff inside the config package.